### DISCUSSION TITLE
How to communicate with a serial device using Avalonia?

### DISCUSSION THREAD

**Discussion by Zopenzop** [AvaloniaUI/Avalonia#8937] [Q&A] [answered]
I have been using Flutter all this time to make an android app to suit my needs. Now, I am well versed with WPF having used it for a long time and having to learn a new framework just to make one app wasn't exactly the best experience.  However, achieving Serial Communication using some framework like React Native or Flutter seemed extremely welcoming since I didn't have much time to spend and these frameworks already had community-made packages for the same. But I couldn't find anything as such for Avalonia. I am already using C#, XAML and WPF to make my Windows app and would absolutely love to use C# and XAML to make my android apps as well.

**How do I achieve Serial Communication for Android using Avalonia? Is it the same as serial communication on Windows? `System.IO.Ports`? `SerialPort.GetPortNames()`?**

I am pretty short on time so instead of experimenting I would like to know if anyone has previously tried going for something similar using Avalonia and if there are some good resources to get started with it. Thanks!

**Comment by kekekeks** [comment, marked as answer]
Communicating with OS services is outside of the scope of Avalonia.

You need to use the underlying API which in this case is Xamarin.Android. I think this library might be helpful - https://github.com/anotherlab/UsbSerialForAndroid

**Reply by Zopenzop** [reply]
@kekekeks That does seem like a good point to get started with! Thanks!
